### PR TITLE
AlignTest / BinaryHeapTest / SortTest

### PR DESCRIPTION
**Pull request by JujuG02**
I added tests for the classes Align, BinaryHeap and Sort in the package com.badlogic.gdx.utils.

### CODE REVIEW

**Review by Frosty-J — CHANGES_REQUESTED** [review]
More util classes I've never used! Not sure why there's so many contributors in this pull request... hello, everyone. It could be cool to check that sorting hasn't exploded - I will let one of the seniors decide.

In the meantime, there's a few things marring the quality of this pull request such as the inclusion of seemingly unrelated files. In future, I suggest keeping your `master` branch synched with `libgdx:master` and creating new branches from this. So you could have `utils-tests` and `tiled-tests` branches to keep matters separated, which can be safely deleted once they're no longer being developed on.

It's great that you're getting involved, though!

File: tests/gdx-tests-android/assets/data/maps/tiled-groups-imagelayer/Lonesome_Forest_Summer_Tileset 2.tsx
[the remark this answers is not in the document]
Comment (BoBIsHere86) [reply tests/gdx-tests-android/assets/data/maps/tiled-groups-imagelayer/Lonesome_Forest_Summer_Tileset 2.tsx:1]: I think these changes should be removed from this PR. Unless I'm missing something they have nothing to do with the newly added tests.
Also note to James, the original version of the tiled test files were added back in PR 7135 which has been merged. I'll make a note to myself to sneak in a change to name of the tileset in one of the next Tiled PRs we have lined up. Leaving that as the name was an oversight from the original contributor from a file I gave him to test.

Also it's odd they are all duplicated with a 2 at the end. I'm guessing a mistake?

[the remark this answers is not in the document]
Comment (Frosty-J) [reply tests/gdx-tests-android/assets/data/maps/tiled-groups-imagelayer/Lonesome_Forest_Summer_Tileset 2.tsx:1]: The Tiled legend is here! Thanks, I saw the non-2 versions of the file in the fork but my monkey brain didn't register that this must mean they're in the main libGDX repo as well, not freshly introduced here.

[the remark this answers is not in the document]
Comment (JujuG02) [reply tests/gdx-tests-android/assets/data/maps/tiled-groups-imagelayer/Lonesome_Forest_Summer_Tileset 2.tsx:1]: Hello ! It's a mistake, I don't really know why there are duplicates but I deleted them.

File: gdx/test/com/badlogic/gdx/utils/BinaryHeapTest.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.utils;
+
+import static org.junit.Assert.*;
+import org.junit.Before;
+import org.junit.Test;
+
+public class BinaryHeapTest {
```
[the remark this answers is not in the document]
Comment (JujuG02) [reply gdx/test/com/badlogic/gdx/utils/BinaryHeapTest.java:8]: Thanks, I didn't see this class. So BinaryHeapTest isn't really useful.

File: gdx/test/com/badlogic/gdx/utils/AlignTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+ * Copyright 2024 Julien Abraul Guilherme.
```
[the remark this answers is not in the document]
Comment (JujuG02) [reply gdx/test/com/badlogic/gdx/utils/AlignTest.java]: I deleted all the copyright then !

File: build.gradle [hunk 3]
```diff
@@ -18,6 +18,7 @@ buildscript {
 
 	repositories {
 		mavenCentral()
+		mavenLocal()
```
[the remark this answers is not in the document]
Comment (JujuG02) [reply build.gradle:21]: Yes, we used it when we where trying to build the LibGDX project at home.

File: gdx/test/com/badlogic/gdx/utils/BinaryHeapTest.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
+/*******************************************************************************
+ * Copyright 2024 AUTHOR
```
Comment (Frosty-J) [inline gdx/test/com/badlogic/gdx/utils/BinaryHeapTest.java:2]: See https://github.com/search?q=repo%3Alibgdx%2Flibgdx+%22see+authors+file%22&type=code for examples of what it looks like. The other classes are better but missing the capital S.

Comment (JujuG02) [reply gdx/test/com/badlogic/gdx/utils/BinaryHeapTest.java:2]: Thank you ! I changed the licences

File: gdx/test/com/badlogic/gdx/utils/AlignTest.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+	@Test
+	public void testToString () {
+		/*
+		 * This method tests if the toString function correctly converts the alignment constants to their string representations,
+		 * such as "top,left", "bottom,right", and "center,center"
+		 */
+		assertEquals("top,left", Align.toString(Align.topLeft));
+		assertEquals("bottom,right", Align.toString(Align.bottomRight));
+		assertEquals("center,center", Align.toString(Align.center));
```
[the remark this answers is not in the document]
Comment (JujuG02) [reply gdx/test/com/badlogic/gdx/utils/AlignTest.java]: I completed this class. Is it better ?

[the remark this answers is not in the document]
Comment (Frosty-J) [reply gdx/test/com/badlogic/gdx/utils/AlignTest.java]: I think you've now made it more complicated to no real gain. The string representation isn't necessarily set in stone, since no sane programmer would rely on it, but that doesn't mean it's worth writing a test case and worrying about if the commas have spaces or they become dashes or the capitalisation changes or any other number of tweaks some grammar Nazi from the future might make.

[the remark this answers is not in the document]
Comment (JujuG02) [reply gdx/test/com/badlogic/gdx/utils/AlignTest.java]: I added a separator attribute in Align class so that people can set the separator they want and I corrected the tests accordingly

[the remark this answers is not in the document]
Comment (Frosty-J) [reply gdx/test/com/badlogic/gdx/utils/AlignTest.java]: We don't need a new `Align` method. Anyone who wants to change the separator in their own game could override `Align#toString`, which exists more for debug purposes than anything. If your heart is set on modifying non-test classes, I'd consider having that be a separate pull request so as to not increase the likelihood of this one being rejected (an inevitable part of increasing scope). I'm sorry if I've confused you.

File: gdx/src/com/badlogic/gdx/utils/Align.java [hunk 6]
```diff
@@ -30,6 +30,8 @@ public class Align {
 	static public final int bottomLeft = bottom | left;
 	static public final int bottomRight = bottom | right;
 
+	private static String separator = ",";
```
Comment (SimonIT) [inline gdx/src/com/badlogic/gdx/utils/Align.java:33]: I don't think it's necessary to make the separator for the `toString`-method configurable...

File: gdx/src/com/badlogic/gdx/utils/reflect/Method.java [hunk 7]
```diff
@@ -146,4 +146,18 @@ public Annotation getDeclaredAnnotation (Class<? extends java.lang.annotation.An
...
+
+	/** @return true if the method is a bridge method */
+	public boolean isBridge () {
+		return method.isBridge();
+	}
+
+	public int getModifiers () {
+		return method.getModifiers();
+	}
+
```
Comment (SimonIT) [inline gdx/src/com/badlogic/gdx/utils/reflect/Method.java:162]: What does this have to do with _AlignTest / BinaryHeapTest / SortTest_?

File: gdx/test/com/badlogic/gdx/math/EllipseTest.java [hunk 8]
```diff
@@ -0,0 +1,6 @@
+
+package com.badlogic.gdx.math;
+
+public class EllipseTest {
+
+}
```
Comment (SimonIT) [inline gdx/test/com/badlogic/gdx/math/EllipseTest.java:6]: Why is this empty?

**Review by SimonIT — CHANGES_REQUESTED** [review]

File: gdx/src/com/badlogic/gdx/utils/reflect/Method.java (hunk 7 above)
Comment (Frosty-J) [reply gdx/src/com/badlogic/gdx/utils/reflect/Method.java:162]: @SimonIT I thought I was going crazy when I last looked for this and couldn't find it. It appears to be accidentally included from #7517 - you'll see these four users are working together and haven't mastered Git.

Comment (SimonIT) [reply gdx/src/com/badlogic/gdx/utils/reflect/Method.java:162]: Yeah... I thought I had déjà vu^^ The thing is, we will probably run into an unnecessary merge conflict if we merge both PRs
